ui/app/modifiers/code-mirror.ts

```diff
@@ -0,0 +1,59 @@
+import Modifier from 'ember-modifier';
+import codemirror from 'codemirror';
+
+import waypointHclMode from './utils/waypointHclMode';
+import 'codemirror/addon/edit/matchbrackets';
+import 'codemirror/addon/edit/closebrackets';
+import 'codemirror/addon/selection/active-line';
+import 'codemirror/addon/mode/simple';
+
+const _PRESET_DEFAULTS: codemirror.EditorConfiguration = {
+  theme: 'monokai',
+  lineNumbers: true,
+  cursorBlinkRate: 500,
+  matchBrackets: true,
+  autoCloseBrackets: true,
+  styleActiveLine: true,
+};
+interface Args {
+  positional: never;
+  named: {
+    value: string;
+    onInput: (value: string) => void;
+    options: Record<string, unknown>;
+  };
+}
+
+// call to create the CodeMirror waypoint HCL mode for syntax highlighting
+waypointHclMode();
```

Comment:
I think this is fine, though I’d be tempted to call the function something like `registerWaypointHclMode`.

An alternative approach would be to use this kind of import:

```ts
import './utils/waypoint-hcl-mode';
```

And then in the module, it can perform the `defineMode` stuff directly (rather than exporting a function):

```ts
import codemirror from 'codemirror';

codemirror.defineSimpleMode('waypointHCL', {
  // ...
});
```